### PR TITLE
maven: use default jdk

### PR DESCRIPTION
**Pull request by raboof**
###### Motivation for this change

While builds using maven are relatively likely to target jdk8,
they typically support building using default jdk.

The main exception is the default jdk no longer supports targeting
JRE's before version 8. For projects that do that we can override the
jdk version for the maven used there.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by raboof** [comment]
Result of `nixpkgs-review pr 114072` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>exhibitor</li>
    <li>gephi</li>
    <li>global-platform-pro</li>
    <li>maven</li>
  </ul>
</details>

**Comment by taku0** [comment]
`undefined variable 'maven' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-1/pkgs/development/java-modules/junit/default.nix:9:13`

**Comment by raboof** [comment]
> `undefined variable 'maven' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-1/pkgs/development/java-modules/junit/default.nix:9:13`

Oops, funny nixpkgs-review didn't catch that. fixed.

File: pkgs/development/java-modules/maven-hello/default.nix [hunk 1]
```diff
@@ -31,6 +31,11 @@ in rec {
     mavenDeps = [];
     sha512 = "3kv5z1i02wfb0l5x3phbsk3qb3wky05sqn4v3y4cx56slqfp9z8j76vnh8v45ydgskwl2vs9xjx6ai8991mzb5ikvl3vdgmrj1j17p2";
     version = "1.0";
+    mavenbuild = mavenbuild.override {
+      maven = pkgs.maven.override {
+        jdk = pkgs.jdk8;
```
Comment (SuperSandro2000) [inline pkgs/development/java-modules/maven-hello/default.nix:36]: This shouldn't use pkgs as input.

**Comment by omasanori** [comment]
Result of `nixpkgs-review pr 114072` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>python37Packages.skein</li>
    <li>python38Packages.skein</li>
    <li>python39Packages.skein</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>exhibitor</li>
    <li>gephi</li>
    <li>global-platform-pro</li>
    <li>maven (maven3)</li>
    <li>plantuml-server</li>
    <li>soapui</li>
    <li>sourcetrail</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114072` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.skein</li>
<li>python38Packages.skein</li>
<li>python39Packages.skein</li>
</ul>
</details>
<details>
<summary>
7 packages built:
</summary>
<ul>
<li>exhibitor</li>
<li>gephi</li>
<li>global-platform-pro</li>
<li>maven (maven3)</li>
<li>plantuml-server</li>
<li>soapui</li>
<li>sourcetrail</li>
</ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM

**Pull request merged**